ui-lib/src/main/java/co/electriccoin/zcash/ui/screen/scankeystone/view/ScanKeystoneView.kt

```diff
@@ -0,0 +1,654 @@
+                validationResult = validationResult,
+                onOpenSettings = onOpenSettings,
+                scanState = scanState,
+                modifier =
+                    Modifier
+                        .fillMaxWidth()
+                        .padding(
+                            horizontal = 24.dp
+                        )
+            )
+        }
+    }
+}
+
+@Composable
+fun ScanFrame(modifier: Modifier = Modifier) {
+    @Suppress("MagicNumber")
+    Box(
+        modifier = modifier.testTag(ScanTag.QR_FRAME)
+    ) {
+        Icon(
+            imageVector = ImageVector.vectorResource(R.drawable.ic_scan_corner),
+            tint = Color.White,
+            contentDescription = null,
+            modifier =
+                Modifier
+                    .rotate(270f)
+                    .align(Alignment.TopStart),
+        )
+        Icon(
+            imageVector = ImageVector.vectorResource(R.drawable.ic_scan_corner),
+            tint = Color.White,
+            contentDescription = null,
+            modifier =
+                Modifier
+                    .rotate(0f)
+                    .align(Alignment.TopEnd),
+        )
+        Icon(
+            imageVector = ImageVector.vectorResource(R.drawable.ic_scan_corner),
+            tint = Color.White,
+            contentDescription = null,
+            modifier =
+                Modifier
+                    .rotate(180f)
+                    .align(Alignment.BottomStart),
+        )
+        Icon(
+            imageVector = ImageVector.vectorResource(R.drawable.ic_scan_corner),
+            tint = Color.White,
+            contentDescription = null,
+            modifier =
+                Modifier
+                    .rotate(90f)
+                    .align(Alignment.BottomEnd),
+        )
+    }
+}
+
+@OptIn(ExperimentalPermissionsApi::class)
+@Suppress("LongMethod")
+@Composable
+fun ScanCameraView(
+    framePosition: FramePosition,
+    onScanned: (result: String) -> Unit,
+    permissionState: PermissionState,
+    setScanState: (ScanScreenState) -> Unit,
+) {
+    val context = LocalContext.current
+    val lifecycleOwner = LocalLifecycleOwner.current
+
+    // We check the permission first, as the ProcessCameraProvider's emit won't be called again after
+    // recomposition with the permission granted
+    val cameraProviderFlow =
+        if (permissionState.status.isGranted) {
+            remember {
+                flow<ProcessCameraProvider> { emit(ProcessCameraProvider.getInstance(context).await()) }
+            }
+        } else {
+            null
+        }
+
+    val collectedCameraProvider = cameraProviderFlow?.collectAsState(initial = null)?.value
+
+    val cameraController = remember { mutableStateOf<CameraControl?>(null) }
+
+    if (null == collectedCameraProvider) {
+        // Show loading indicator
+    } else {
+        val contentDescription = stringResource(id = R.string.scan_preview_content_description)
```

Comment:
Let's add e.g. `scan_keystone` resource folder and separate these strings into it.

Reply:
✅